File: app/src/main/java/me/ykrank/s1next/view/adapter/simple/SimpleRecycleViewAdapter.java
package me.ykrank.s1next.view.adapter.simple;

import android.content.Context;
import android.support.annotation.LayoutRes;
import android.support.annotation.NonNull;

import me.ykrank.s1next.view.adapter.BaseRecyclerViewAdapter;

/**
 * Simple adapter, just one type item, or {@link me.ykrank.s1next.view.adapter.item.ProgressItem}.
 * Layout databinding variable name should be only "model".
 * Created by ykrank on 2017/3/22.
 */

public class SimpleRecycleViewAdapter extends BaseRecyclerViewAdapter {
    
    public SimpleRecycleViewAdapter(@NonNull Context context, @LayoutRes int layoutRes) {
        this(context, layoutRes, null);
    }

    public SimpleRecycleViewAdapter(@NonNull Context context, @LayoutRes int layoutRes, BindViewHolderCallback bindViewHolderCallback) {
        super(context);
        addAdapterDelegate(new SimpleAdapterDelegate(context, layoutRes, bindViewHolderCallback));
    }


}

File: library/src/main/java/com/github/ykrank/androidtools/ui/adapter/simple/SimpleRecycleViewAdapter.kt
package com.github.ykrank.androidtools.ui.adapter.simple

import android.content.Context
import androidx.databinding.ViewDataBinding
import androidx.annotation.LayoutRes
import com.github.ykrank.androidtools.ui.adapter.LibBaseRecyclerViewAdapter
import com.github.ykrank.androidtools.ui.adapter.delegate.item.ProgressItem
import com.hannesdorfmann.adapterdelegates3.AdapterDelegate


/**
 * Simple adapter, just one type item, or [ProgressItem].
 * Layout databinding variable name should be only "model".
 * Created by ykrank on 2017/3/22.
 */

class SimpleRecycleViewAdapter : LibBaseRecyclerViewAdapter {

    constructor(context: Context, @LayoutRes layoutRes: Int, stableId: Boolean = false, bindViewHolderCallback: BindViewHolderCallback? = null,
                createViewHolderCallback: ((ViewDataBinding) -> Unit)? = null) : super(context, stableId) {
        addAdapterDelegate(SimpleAdapterDelegate(context, layoutRes, null, createViewHolderCallback, bindViewHolderCallback))
    }

    constructor(context: Context, adapterDelegate: AdapterDelegate<MutableList<Any>>, stableId: Boolean = false) : super(context, stableId) {
        addAdapterDelegate(adapterDelegate)
    }
}
